poto/lambda_func.py

```
@@ -0,0 +1,182 @@
+    os.chdir(dir_)
+    
+    postfix = datetime.now().strftime("%Y%m%d%H%M")
+    temp_dir_path =_make_temp_dir(postfix)
+    temp_dir = os.path.basename(temp_dir_path)
+
+    if not is_non_editible_packages:
+        # default exclude에 포함 안 되는 거 제외
+        editible_packages_pattern = _get_editible_packages_pattern(is_non_editible_packages, zappa_stage)
+        _copy_editible_packages(temp_dir_path, editible_packages_pattern)
+    
+    dir_ = (lambda dir_: dir_ if dir_[-1] != '/' else dir_[:-1])(dir_)
+    output_zip = f"{os.path.basename(dir_)}_{postfix}.zip"
+    os.system(f'cp -r `ls -A | grep -vE "{temp_dir}"` {temp_dir}')
+
+    os.chdir(temp_dir_path)
+    os.system(f"zip -r9 {output_zip} *" )
+    os.system(f"chmod 755 {output_zip}")
+
+    zip_dest_path = os.path.join(output_zip_path, output_zip)
+    os.rename(output_zip, zip_dest_path)
+    shutil.rmtree(temp_dir_path)
+
+    print(f'fuction zip {zip_dest_path} created')
+    
+    return zip_dest_path    
+
+def _get_editible_packages_pattern(is_non_editible_packages, zappa_stage):
+    settings = json.load(open("zappa_settings.json"))
+    if not is_non_editible_packages:    
+        if not settings[zappa_stage].get('exclude'):
+            raise AssertionError('editible packages를 zappa_settings exclude에 추가해주세요. ex) "exclude": [your_packs]')
+        editible_packages_pattern = [p for p in settings[zappa_stage]['exclude'] if not re.search("|".join(DEFAULT_EXCLUDE_PACKS), p)]
+        print('editible_packages_pattern', editible_packages_pattern)    
+    else:
+        editible_packages_pattern = None
+    
+    return editible_packages_pattern
+
+def _check_django_mysql_version():
+    try:
+        import django
+        if int(django.__version__) >= 2.2:
+            try:
+                import MySQLdb
+                raise AssertionError('mysqlclient 1.3.13은 lambda에서 쓸 수 없어요...')
+            except:
+                pass
```

try except: pass 에서 raise 가 있는건 의미가 없습니다